src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+
+using SceneTableEntry   = chip::scenes::DefaultSceneTableImpl::SceneTableEntry;
+using SceneStorageId    = chip::scenes::DefaultSceneTableImpl::SceneStorageId;
+using SceneData         = chip::scenes::DefaultSceneTableImpl::SceneData;
+using HandlerContext    = chip::app::CommandHandlerInterface::HandlerContext;
+using ExtensionFieldSet = chip::scenes::ExtensionFieldSet;
+using GroupDataProvider = chip::Credentials::GroupDataProvider;
+using SceneTable        = chip::scenes::SceneTable<chip::scenes::ExtensionFieldSetsImpl>;
+using AuthMode          = chip::Access::AuthMode;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Scenes {
+
+/// @brief Generate and add a response to a command handler context depending on an EmberAfStatus
+/// @tparam ResponseType Type of response, depends on the command
+/// @param ctx Command Handler context where to add reponse
+/// @param resp Response to add in ctx
+/// @param status Status to verify
+/// @return EmberAfStatus -> CHIP_ERROR
+template <typename ResponseType>
+CHIP_ERROR AddResponseOnError(CommandHandlerInterface::HandlerContext & ctx, ResponseType & resp, EmberAfStatus status)
+{
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        resp.status = to_underlying(ToInteractionModelStatus(status));
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+    }
+    return StatusIB(ToInteractionModelStatus(status)).ToChipError();
+}
+
+/// @brief Generate and add a response to a command handler context if err parameter is not CHIP_NO_ERROR
+/// @tparam ResponseType Type of response, depends on the command
+/// @param ctx Command Handler context where to add reponse
+/// @param resp Response to add in ctx
+/// @param status Status to verify
+/// @return CHIP_ERROR
+template <typename ResponseType>
+CHIP_ERROR AddResponseOnError(CommandHandlerInterface::HandlerContext & ctx, ResponseType & resp, CHIP_ERROR err)
+{
+    if (CHIP_NO_ERROR != err)
+    {
+        resp.status = to_underlying(StatusIB(err).mStatus);
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+    }
+    return err;
+}
+
+ScenesServer ScenesServer::mInstance;
+
+ScenesServer & ScenesServer::Instance()
+{
+    return mInstance;
+}
+void ReportAttributeOnAllEndpoints(AttributeId attribute) {}
+
+CHIP_ERROR ScenesServer::Init()
+{
+    // Prevents re-initializing
+    VerifyOrReturnError(!mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(chip::app::InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+    mGroupProvider = Credentials::GetGroupDataProvider();
+
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable = chip::scenes::GetSceneTableImpl();
+    ReturnErrorOnFailure(sceneTable->Init(&chip::Server::GetInstance().GetPersistentStorage()));
+
+    for (auto endpoint : EnabledEndpointsWithServerCluster(Id))
```

Comment:
This does not look right.  If the idea is to init the scenes cluster on the endpoints where it should be inited, we have a callback for that, which has the virtue of being called correctly when dynamic endpoints are added and whatnot: `emberAfScenesClusterServerInitCallback(chip::EndpointId endpoint)`